### DISCUSSION TITLE
NLP Parsing Question(s)

### DISCUSSION THREAD

**Discussion by joel-madero** [mealie-recipes/mealie#4635] [Q&A] [answered]
Hi All,

I'm trying to figure out how to use mealie "the right way" and I keep coming back to the natural language parser. I'm on most recent version of mealie (2.3.0). What I can't see is how this actually works. In the debug mode I can see that it does cool things based apparently on a NYT list but in actual recipes, I don't see the same behavior. 

I even played around in the test environment created by the dev(s) and I don't see it working. For example, I created a new grocery list called "Testing" (everyone should be able to see it), and ingredients are being matched up so I see duplicates based on different recipes being added:

https://demo.mealie.io/shopping-lists/9c3bfe84-935e-4217-bb1a-35f5759c9c3d

You can see in the list that I have added two recipes to list "Testing" from the test environment: 
(1) Red Chile Tostadas with Eggs 
(2) Quick Baked Chicken Parmesan

Both have eggs. In the shopping list I have two different eggs listed:
(1) 6 eggs
(2) 1 egg, beaten

Am I missing something?

**Comment by michael-genson** [comment, marked as answer]
https://docs.mealie.io/documentation/getting-started/faq/#how-do-i-enable-smart-ingredient-handling

**Comment by joel-madero** [comment]
Thanks that helped but couple questions:
(1) There's no way to do this more efficiently than one recipe at a time?
(2) Is there a way to turn this feature on automatically for all new recipes (import and manual creation)?

**Reply by michael-genson** [reply]
Currently the only way to parse is one recipe at a time. Right now the parser isn't reliable enough to fully automate, so it requires some human intervention. I typically parse recipes as I make them

You can enable parsing by default on recipes in your household settings, however it doesn't auto-parse, it simply enables parsing

**Comment by joel-madero** [comment]
Gotcha. What might be nice is to make an option for all new recipes manually entered start in the "parse mode" or whatever it's called. (i.e., with "Disable Ingredient Amounts" disabled by default). This way when at least manually entering, it starts in the preferred mode for the user without needing to manually deselect every time. Just a thought. Thanks for the assistance.

**Reply by michael-genson** [reply]
You can do this from your household settings:
![image](https://github.com/user-attachments/assets/6f6485ac-724b-4e55-b287-411a9f3a9477)

**Reply by amatriain** [reply]
I can't see this option in mealie 3.2.0 , has it been removed?
